### DISCUSSION TITLE
怎么在使用gemini plugin的时候，使用梯度累计

### DISCUSSION THREAD

**Discussion by bisunny** [hpcaitech/ColossalAI#4610] [Community | Q&A] [answered]
我在尝试跑llama2的例子时发现没有梯度累计的代码，而例子中（gemini.sh）是用的gemini，请问这个例子怎么开启梯度累计，把llama每个global batch的token数量调整到4M

**Comment by sherlcok314159** [comment]
Gemini 是基于Chunk内存管理和异构内存管理的 Zero-3，它不支持局部梯度累积

**Comment by Fridge003** [comment, marked as answer]
您好，gemini对梯度累积的支持已经完成。
使用方法可以参考[这个文档](https://colossalai.org/docs/features/gradient_accumulation_with_booster)。

**Reply by bisunny** [reply]
好的，感谢

**Reply by airlsyn** [reply]
@Fridge003  参考此文档的 Gradient Accumulation on GeminiPlugin，会在 booster.backward(train_loss, gemini_optimizer) 遇到 `AssertionError: you are calculating the l2 norm twice`

**Reply by Fridge003** [reply]
已在 #5002 修复
